### PR TITLE
For #668: Restyles custom tabs

### LINKED ISSUE #668: Update Custom Tabs to Match UX
Custom tabs were restyled in #203 to have a top app bar with developer-provided actions and to appear substantially different from the normal Fenix browser fragment. A complete restyling of our browser fragment will be required for custom tabs.

### PR DESCRIPTION
**Pull request by sblatz**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 1]
```diff
@@ -130,15 +131,25 @@ class BrowserFragment : Fragment(), BackHandler, CoroutineScope {
...
+                gravity = getAppropriateLayoutGravity()
+
                 val accessibilityManager = context
                     ?.getSystemService(Context.ACCESSIBILITY_SERVICE) as AccessibilityManager
 
-                if (!accessibilityManager.isTouchExplorationEnabled) {
-                    behavior = BrowserToolbarBottomBehavior(view.context, null)
+                behavior = if (gravity == Gravity.TOP) {
+                    AppBarLayout.ScrollingViewBehavior()
+                } else if (!accessibilityManager.isTouchExplorationEnabled) {
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: Please delete this whole else branch (see `setToolbarBehavior`)

File: app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 2]
```diff
@@ -255,20 +272,24 @@ class BrowserFragment : Fragment(), BackHandler, CoroutineScope {
...
-                    id,
-                    requireActivity(),
-                    onItemTapped = { actionEmitter.onNext(SearchAction.ToolbarMenuItemTapped(it)) }
-                ),
-                owner = this,
-                view = view
-            )
+
+        sessionId?.let { sessionId ->
+            if (sessionManager.findSessionById(sessionId)?.isCustomTabSession() == true) {
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: Sorry about the drive-by comment: If the `sessionId` is not null, mean that the session is a Custom Tabs session without needing to re-verify. Did you see this check working for you rather than the `sessionId` check? That sounds like our Custom Tab detection isn't working accurately (which we've seen act oddly depending on the app calling it).

Comment (sblatz) [reply app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: When I remove this check, the title view then displays in normal tabs. A normal session will reach this part of `BrowserFragment`, and we need to only set this `customTabIntegration` if it's a custom tab.

Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt]: This sounds like a bug in that case. I've filed https://github.com/mozilla-mobile/fenix/issues/1639 as a follow up.
